### PR TITLE
IN-5 feat: add pagination

### PR DESCRIPTION
**Pull request by web-serge**

### CODE REVIEW

File: src/components/pagination/pagination.module.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      padding: 0 6px;
+      background: var(--color-dark-300);
+    }
+  }
+
+  &__button {
+    padding: 6px 8px;
+    color: var(--color-light-100);
+    &:hover {
+      color: white;
```
[the remark this answers is not in the document]
Comment (web-serge) [reply src/components/pagination/pagination.module.scss:61]: fix

File: src/components/pagination/pagination.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import type { Meta, StoryObj } from "@storybook/react"
+
+import { useState } from "react"
+
+import { Pagination } from "./pagination"
+
+const meta = {
+  component: Pagination,
+  parameters: {
```
[the remark this answers is not in the document]
Comment (web-serge) [reply src/components/pagination/pagination.stories.tsx:9]: fix

File: src/components/pagination/pagination.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+export const Control = {
+  args: {
+    currentPage: 1,
+    pageSize: 100,
+    paginationOptions,
+    siblingCount: 1,
+    totalCount: 1001
+  },
+  render: function Render(args: any) {
```
[the remark this answers is not in the document]
Comment (web-serge) [reply src/components/pagination/pagination.stories.tsx:40]: How is this an option? "ComponentProps<typeof Pagination>"

File: src/components/pagination/pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  }
+  const prevPage = () => {
+    onPageChange(currentPage - 1)
+  }
+  const paginationRange = usePagination({ currentPage, pageSize, siblingCount, totalCount })
+
+  return (
+    <div className={s.pagination}>
+      <Button
+        className={s.pagination__button}
```
[the remark this answers is not in the document]
Comment (web-serge) [reply src/components/pagination/pagination.tsx:50]: fix

**Review by SergFediaev — APPROVED** [review]

**Review by vitaliiRss — APPROVED** [review]

**Review by Abrosimov96 — APPROVED** [review]

**Pull request merged**